### ISSUE TITLE
Suggestion: a way to enforce CONSTANT_CASE for constants

### ISSUE THREAD

**Issue by thealjey** [eslint/eslint#4175]
It would really be great if eslint could enforce the use of CONSTANT_CASE for constants, e.g:
```javascript
// Bad
const someValue = 10;

// Good
const SOME_VALUE = 10;
```
And similarly - camelCase for variables:
```javascript
// Bad
let SOME_VALUE = 10;

// Good
let someValue = 10;
```

**Comment by eslintbot** [comment]
Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by platinumazure** [comment]
Would `id-match` do what you want? I can't remember if it distinguishes
between `var`/`let`/`const`.

**Comment by thealjey** [comment]
I honestly do not think it would, as I don't need to check all identifiers, but only constant definitions.

**Comment by nzakas** [comment]
The `camelcase` rule basically does the second one already. I don't think we can introduce a new rule for the first one because it could overlap with `camelcase` . maybe an option on `camelcase`?

**Comment by thealjey** [comment]
Again, `camelcase` enforces it's style on all identifiers, not just variable definitions, but on functions, class names, imported variables (which frankly does not make any sense to me), etc.
Furthermore, it does not distinguish constants and variables.
As to creating a new rule vs adding an option to an existing one - I don't really know or care about that :smile:

**Comment by nzakas** [comment]
I'm sorry, you'll need to create a custom rule for this. What you're asking for would overlap would several rules, and we don't do that as it causes confusion.
